src/modules/utils/motordrivercontrol/MotorDriverControl.h

```diff
@@ -38,20 +40,34 @@ class MotorDriverControl : public Module {
 
         void enable(bool on);
         int sendSPI(uint8_t *b, int cnt, uint8_t *r);
-
-        Pin spi_cs_pin;
-        mbed::SPI *spi;
+        int sendUART(uint8_t *b, int cnt, uint8_t *r);
+        
+        // use SPI or UART
+        union {
+            struct {
+                Pin *spi_cs_pin;
+                mbed::SPI *spi;
+            };
+            struct {
+                Pin *sw_uart_tx_pin;
+                Pin *sw_uart_rx_pin;
+                bool setup;
```

Comment:
remove the setup and serial from this union and the sizes will be the same and use the same amount of memory as before.

Reply:
ok, so i will keep uart pins inside a struct and also spi pin